File: DealApp.API/Models/Property.cs
using System;
using System.Collections.Generic;

namespace DealApp.API.Models
{
    public class Property
    {
        public int Id { get; set; }
        public string PropertyName { get; set; }
        public int? CategoryId { get; set; }
        public int? PropertyTypeId { get; set; }
        public int? CityId { get; set; }
        public int? QuarterId { get; set; }
        public int? ProjectId { get; set; }
        public int? TowerId { get; set; }
        public int? BlockId { get; set; }
        public int? AreaId { get; set; }
        public int? InterfaceId { get; set; }
        public string PropertyNumber { get; set; }
        public decimal? Price { get; set; }
        public int? RenterId { get; set; }
        public string Note { get; set; }
        public bool? IsAvailable { get; set; }
        public bool? IsActive { get; set; }
        public bool? IsDeleted { get; set; }
        public string CreatedBy { get; set; }
        public DateTime? CreatedDate { get; set; }
        public string ModifyBy { get; set; }
        public DateTime? ModifyDate { get; set; }
        public bool? IsFocusedOn { get; set; }
        public bool? ItHasAproblem { get; set; }
        public bool? IsFurnished { get; set; }
        public ICollection<Photo> Photos { get; set; }


        public virtual Area Area { get; set; }
        public virtual Block Block { get; set; }
        public virtual Category Category { get; set; }
        public virtual City City { get; set; }
        public virtual Floor Floor { get; set; }
        public virtual Interface Interface { get; set; }
        public virtual Project Project { get; set; }
        public virtual PropertyType PropertyType { get; set; }
        public virtual Tower Tower { get; set; }
        public virtual Quarter Quarter { get; set; }
        public virtual Renter Renter { get; set; }
    }   
}

File: DealApp-SPA/src/app/_models/property.ts
import { City } from './city';
import { Photo } from './photo';


export interface Property {
    id: number;
    propertyName: string;
    photoUrl: string;
    cityName: string;
    quarterName: string;
    towerName: string;
    areaName: string;
    blockName: string;
    categoryName: string;
    floorName: string;
    interfaceName: string;
    projectName: string;
    propertyTypeName: string;
    renterName: string;
    renterMobileNo1?: number;
    renterMobileNo2?: number;
    categoryId?: number;
    propertyTypeId?: number;
    projectId?: number;
    towerId?: number;
    blockId?: number;
    areaId?: number;
    interfaceId?: number;
    propertyNumber: string;
    price?: number;
    renterId?: number;
    note?: string;
    isAvailable: boolean;
    isActive?: boolean;
    isDeleted?: boolean;
    createdBy?: string;
    createdDate?: Date;
    modifiedBy?: string;
    modifiedDate?: Date;
    isFocusedOn?: boolean;
    itHasProblem?: boolean;
    isFurnished?: boolean;
    photos?: Photo[];
    city?: City[];
}



